### PR TITLE
Network api requests

### PR DESCRIPTION
**Pull request by grus4**
Added requestInterception and apiTests specs

### CODE REVIEW

File: tests/requestInterception.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        await page.route('/api/users/profile', async route => {
+            await route.fulfill({
+                status: 200,
+                json: mockedProfileData,
+            });
+        });
+        await garagePage.openProfilePage();
+        await expect(profilePage.profileName).not.toHaveText(
+            `${USERS.USER1.firstName} ${USERS.USER1.lastName}`
+        );
```
Comment (sttaran) [inline tests/requestInterception.spec.js]: redundant assertion

you can remove it

File: src/pageObject/garagePage/GaragePage.js [hunk 2]
```diff
@@ -7,10 +7,14 @@ export default class GaragePage extends BasePage {
...
+        this.editProfileButton = page.locator('.sidebar_btn.-profile');
     }
 
     async openAddCarForm() {
         await this.addCarButton.click();
     }
 
+    async openProfilePage() {
+        await this.editProfileButton.click();
+    }
```
Comment (sttaran) [inline src/pageObject/garagePage/GaragePage.js]: it should not be a part of Garage page

File: tests/requestInterception.spec.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        profilePage,
+        page,
+    }) => {
+        await page.route('/api/users/profile', async route => {
+            await route.fulfill({
+                status: 200,
+                json: mockedProfileData,
+            });
+        });
+        await garagePage.openProfilePage();
```
Comment (sttaran) [inline tests/requestInterception.spec.js]: do 
await profilePage.navigate()

**Review by sttaran — CHANGES_REQUESTED** [review]

File: tests/requestInterception.spec.js (hunk 3 above)
Comment (grus4) [reply tests/requestInterception.spec.js]: got it

Comment (grus4) [reply tests/requestInterception.spec.js]: got it

**Pull request merged**